### PR TITLE
API docs

### PR DESCRIPTION
**Pull request by nonsleepr**
Solves #3.

To see how it looks like, Import URL `https://raw.githubusercontent.com/nonsleepr/python-eufy-security/api-docs/openapi.yaml` into https://editor.swagger.io/ (a bonus: you could actually poke around API there).

The way I see it deployed, is something like described [here](https://github.com/peter-evans/swagger-github-pages).

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,1044 @@
...
+                    example: security-app.eufylife.com
+        '401':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/401Response'
+  /app/upload_hub_params:
+    post:
+      summary: AndroidApp - Set Hum Parameters
```
Comment (phiali) [inline openapi.yaml]: I think this might need to be `Hub` parameters instead of `Hum`?

**Comment by FuzzyMistborn** [comment]
Looks nice!  Glad we finally are getting somewhere.

**Comment by bachya** [comment]
@nonsleepr Now that `dev` is our default branch, please make sure this PR is pointed off of that. Thanks!

**Comment by nonsleepr** [comment]
So, how do you think we should host the UI? Just noting that it could be imported into `editor.swagger.io` (it can't import by crafting the URL right now)? Or shall we utilize Github Pages?

**Comment by bachya** [comment]
@nonsleepr I think GitHub Pages would be nice. 👍

**Comment by nonsleepr** [comment]
Can you set it up? Something like this: https://github.com/peter-evans/swagger-github-pages.

**Comment by bachya** [comment]
Might need @FuzzyMistborn to do it since he owns the repo.

**Comment by FuzzyMistborn** [comment]
Ok the github page is setup, but I can't get it to point to dev (never dealt with swagger or github pages before).  If it needs to go to master you should just commit it and everything _should_ be set up correctly.

**Comment by bachya** [comment]
Done. @FuzzyMistborn, going forward, let's make sure that PRs are squash-merged; it'll ensure we keep a clean commit history.

**Comment by FuzzyMistborn** [comment]
Yeah sorry, been a long week and I probably shouldn't have done that so late.  Thanks!

**Comment by bachya** [comment]
No worries! I’ve been there. 😂👍🏻

**Comment by FuzzyMistborn** [comment]
Hmm....page appears to be rendering blank.  Must be other things we need to change?  I think everything is set up correctly on the backend.

image

**Comment by bachya** [comment]
I'm unclear on whether that URL matches the directory structure in `master` – @nonsleepr, can you assist here?

**Comment by nonsleepr** [comment]
Yeah, it expects to see `index.html` and the contents of `dist` in the same folder.

Possible ways to solve that:

* A. Move stuff from `dist/` to the root (I dislike polluting the root directory though).
* B. Edit `src`s and `href`s in `index.html` to ./dist/???`.
* C. Rename `dist/` to `docs/` and change GH-Pages settings to build from "master branch /docs folder".
* D. Create a branch called `gh-pages` and keep there only the `index.html`, `openapi.yaml` and the *contents* of `dist/` folder; and change GH-Pages settings to build from `gh-pages` branch.

I would prefer the option **D**, it should be easy to automate as well.

**Comment by bachya** [comment]
I like option D, too.

**Comment by FuzzyMistborn** [comment]
Ah, i misread the setup docs.  They said move index.html out of the dist folder....i think that's where I got confused.

I also like option D.

**Comment by bachya** [comment]
I'll take care of it. 👍

**Comment by bachya** [comment]
Okay, `gh-pages` now exists and has the files @nonsleepr mentioned.

**Comment by FuzzyMistborn** [comment]
Updated the settings. On mobile atm, can check later if it worked.

**Comment by nonsleepr** [comment]
It works!

**Pull request merged**